executor/write.go

```diff
@@ -92,7 +93,11 @@ func updateRecord(ctx context.Context, sctx sessionctx.Context, h kv.Handle, old
 
 	// 3. Compare datum, then handle some flags.
 	for i, col := range t.Cols() {
+		collation := newData[i].Collation()
+		// We should use binary collation to compare datum, otherwise the result will be incorrect.
+		newData[i].SetCollation(charset.CharsetBin)
```

Comment:
prefer `chasert.CollationBIn` although it's the same string.

Reply:
Good idea. It will be more clear.